src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDbOptional = userService.findByEmail(email);
+        User user = userFromDbOptional.get();
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
```

Comment:
Проблема не в умові, а в тому що ти робиш перед умовою, ти витягуєш юзера без перевірки чи він там присутній.
Наприклад юзера немає в базі і в тебе буде наступна ситуація.
```suggestion
        User user = userFromDbOptional.get(); метод get() поверне null і буде user = null
        String hashedPassword = HashUtil.hashPassword(password, user.getSalt()); тут ти кажеш null.getSalt() буде помилка.
```

Reply:
Do you mean that if all the same would be in a condition the NullPointerException wouldn't be?